src/components/protocol_handler/src/protocol_handler_impl.cc

```diff
@@ -1120,7 +1124,9 @@ RESULT_CODE ProtocolHandlerImpl::HandleControlMessageStartSession(
                                                   connection_key);
     security_manager::SSLContext* ssl_context =
         GetSSLContextBySession(session_info);
-
+    if (!ssl_context) {
+      ssl_context = security_manager_->CreateSSLContext(connection_key);
```

Comment:
@indlu seting protection flag should be executed when we sure that handshake done shouldn't we?